ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -215,96 +169,84 @@ public void showInterstitial() {
-        Activity activity = (Activity) context;
-
-        // Even though we are a banner request, we still need to initialize UnityAds.
-        UnitySingleton.getInstance().initializeUnityAds(activity, gameId);
+    }
 
-        float density = context.getResources().getDisplayMetrics().density;
-        int bannerWidth = Math.round(adSize.getWidthInPixels(context) / density);
-        int bannerHeight = Math.round(adSize.getHeightInPixels(context) / density);
+    @Override
+    public void onUnityAdsReady(String placementId) {
+        // Unity Ads is ready to show ads for the given placementId.
+    }
 
-        UnityBannerSize size = new UnityBannerSize(bannerWidth, bannerHeight);
+    @Override
+    public void onUnityAdsStart(String placementId) {
+        // Unity Ads video ad started playing. Google Mobile Ads SDK does not support
+        // callbacks for Interstitial ads when they start playing.
+    }
 
-        if (mBannerView == null){
-            mBannerView = new BannerView((Activity)context, bannerPlacementId, size);
+    @Override
+    public void onUnityAdsClick(String s) {
+        // Unity Ads ad clicked.
+        if (mMediationInterstitialListener != null) {
+            mMediationInterstitialListener.onAdClicked(UnityAdapter.this);
+            // Unity Ads doesn't provide a "leaving application" event, so assuming that the
+            // user is leaving the application when a click is received, forwarding an on ad
+            // left application event.
+            mMediationInterstitialListener.onAdLeftApplication(UnityAdapter.this);
         }
-
-        mBannerView.setListener(this);
-        mBannerView.load();
     }
 
     @Override
-    public View getBannerView() {
-        return mBannerView;
+    public void onUnityAdsPlacementStateChanged(String placementId,
+                                                UnityAds.PlacementState oldState,
+                                                UnityAds.PlacementState newState) {
+        // This callback is not forwarded to Google Mobile Ads SDK. onUnityAdsError should be used
+        // to forward Unity Ads SDK state to Google Mobile Ads SDK.
     }
 
     @Override
-    public void onBannerLoaded(BannerView bannerView) {
-        Log.v(TAG, "Unity Ads Banner finished loading banner for placement: " + mBannerView.getPlacementId());
-        bannerListener.onAdLoaded(UnityAdapter.this);
+    public void onUnityAdsFinish(String placementId, UnityAds.FinishState finishState) {
+        // Unity Ads ad closed.
+        if (mMediationInterstitialListener != null) {
+            mMediationInterstitialListener.onAdClosed(UnityAdapter.this);
+        }
     }
 
     @Override
-    public void onBannerClick(BannerView bannerView) {
-        bannerListener.onAdClicked(UnityAdapter.this);
+    public void onUnityAdsError(UnityAds.UnityAdsError unityAdsError, String placementId) {
+        // Unity Ads ad failed to show.
+        if (placementId.equals(getPlacementId()))
```

Comment:
@serge-14 Do we need this check, if so should I add it to onUnityAdsFinish since placement id is a parameter?

Reply:
no... that is not gonna be a `placementId`...

Reply:
what's not gonna be a placement id?

Reply:
`String placementId` is `String errorMessage` actually